sysdig/resource_sysdig_secure_cloud_auth_account.go

```diff
@@ -396,19 +396,92 @@ func cloudauthAccountFromResourceData(data *schema.ResourceData) *v2.CloudauthAc
 	}
 }
 
-func cloudauthAccountToResourceData(data *schema.ResourceData, cloudAccount *v2.CloudauthAccountSecure) error {
-	for _, err := range []error{
-		data.Set("id", cloudAccount.Id),
-		data.Set("enabled", cloudAccount.Enabled),
-		data.Set("cloud_provider_id", cloudAccount.ProviderId),
-		data.Set("cloud_provider_type", cloudAccount.Provider.String()),
-		data.Set("components", cloudAccount.Components),
-		data.Set("feature", cloudAccount.Feature),
-	} {
-		if err != nil {
-			return err
+/*
+	This helper function converts feature values from *cloudauth.AccountFeature to resource data schema.
+*/
+
+func featureValuesToResourceData(name string, feature *cloudauth.AccountFeature) map[string]interface{} {
+	valuesMap := make(map[string]interface{})
+
+	valuesMap["type"] = feature.Type.String()
+	valuesMap["enabled"] = feature.Enabled
+	valuesMap["components"] = feature.Components
+
+	featureMap := map[string]interface{}{
+		name: []map[string]interface{}{
+			valuesMap,
+		},
+	}
+
+	return featureMap
+}
+
+/*
+	This helper function converts the features data from *cloudauth.AccountFeatures to resource data schema.
+	This is needed to set the value in cloudauthAccountToResourceData().
+*/
+
+func featureToResourceData(features *cloudauth.AccountFeatures) []map[string]interface{} {
+	featureMap := []map[string]interface{}{}
+
+	featureFields := map[string]*cloudauth.AccountFeature{
+		"secure_threat_detection":     features.SecureThreatDetection,
```

Comment:
all these strings in quotes, here and below, should we have them defined as constants above?